### ISSUE TITLE
Without woodstox: Empty xml elements is with an endElement-tag

### ISSUE THREAD

**Issue by allanbrohansen** [jamesagnew/hapi-fhir#914]
We have just moved from 2.2 to 3.3 and noticed that all empty elements now uses a end-element.
E.g.
`<name value="HAPI FHIR Server"/>`
has become:
`<name value="HAPI FHIR Server"></name>`

Both are equally valid but we prefer the old shorter version.

It turns out that the difference happens because woodstox-stax is no longer included with HAPI (and we ourselves did not include it).
We have therefore gotten the old behaviour back by manually including woodstox (gradle-syntax):
`compile group: 'org.codehaus.woodstox', name: 'woodstox-core-asl', version: '4.4.1'`

We however like to have the empty element behavior without having to include woodstox.

**Comment by jamesagnew** [comment]
The woodstox dependency was made optional in hapi-fhir-base's pom.xml file, as it was causing issues for Android users (having Woodstox on your classpath seems to prevent Android apps from working at all).

I'm not sure I'm getting your concern? Can't you just add Woodstox to your pom?

**Comment by allanbrohansen** [comment]
Yes, we did add it - and we are happy again, functionality-wise.
I would just like to keep our dependencies as clean as possible, and therefore adding woodstox "just" to have empty elements is a bit annoying.
So ideally I think hapi's XmlWriter should use writeEmptyElement in stead of writeEndElement - when appropriate.

**Comment by jamesagnew** [comment]
Technically it isn't just for the empty elements, Woodstox is a faster and more full featured parser. Unless you have a specific reason not to, I'd highly recommend using it.

Are you interested in tackling the writeEmptyElements thing? I'm hesitant to leave this open, as there are a limited number of hours in the day and fixing something that isn't actually technically broken doesn't seem like the best use of them unless someone is willing to take this bug on.
